### PR TITLE
Sanitize schema database

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
https://improbableio.atlassian.net/browse/UNR-4694

#### Release note
Fixes transient issue introduced in v15 -- not required.

#### Tests
Ran tests generating schema after deleting Unreal object classes.

#### Primary reviewers
@improbable-valentyn

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp [hunk 1]
```diff
@@ -1447,6 +1447,52 @@ bool SpatialGDKGenerateSchemaForClasses(TSet<UClass*> Classes, FString SchemaOut
 	return true;
 }
 
+template <class T>
+void SanatizeClassMap(TMap<FString, T>& Map, const TSet<FName>& ValidClassNames)
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp:1451]:
```suggestion
void SanitizeClassMap(TMap<FString, T>& Map, const TSet<FName>& ValidClassNames)
```

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp:1451]: oh god - it's everywhere

File: SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp [hunk 2]
```diff
@@ -1447,6 +1447,52 @@ bool SpatialGDKGenerateSchemaForClasses(TSet<UClass*> Classes, FString SchemaOut
...
+{
+	TSet<FString> ItemsToRemove;
+
+	for (const auto& Item : Map)
+	{
+		FString SanatizeName = Item.Key;
+		SanatizeName.RemoveFromEnd(TEXT("_C"));
+		if (ValidClassNames.Find(FName(SanatizeName)) == nullptr)
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp:1459]:
```suggestion
		if (!ValidClassNames.Contains(FName(SanatizeName)))
```

Comment (KeyMaster-) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp:1459]:
```suggestion
		if (!ValidClassNames.Contains(FName(SanatizeName)))
```

**Review by improbable-valentyn — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp [hunk 3]
```diff
@@ -1447,6 +1447,52 @@ bool SpatialGDKGenerateSchemaForClasses(TSet<UClass*> Classes, FString SchemaOut
 	return true;
 }
 
+template <class T>
+void SanitizeClassMap(TMap<FString, T>& Map, const TSet<FName>& ValidClassNames)
+{
+	TSet<FString> ItemsToRemove;
+
+	for (const auto& Item : Map)
```
Comment (ImprobableNic) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp:1455]: Could avoid the extra set and use Iterator.RemoveCurrent()

**Pull request merged**